substrate/runtime/executive/src/lib.rs

```diff
@@ -0,0 +1,277 @@
+// Copyright 2017 Parity Technologies (UK) Ltd.
+// This file is part of Substrate Demo.
+
+// Substrate Demo is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate Demo is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate Demo.  If not, see <http://www.gnu.org/licenses/>.
+
+//! System manager: Handles all of the top-level stuff; executing block/transaction, setting code
+//! and depositing logs.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg_attr(test, macro_use)] extern crate substrate_runtime_std as rstd;
+extern crate substrate_runtime_support as runtime_support;
+extern crate substrate_runtime_io as runtime_io;
+extern crate substrate_codec as codec;
+extern crate substrate_runtime_primitives as primitives;
+extern crate substrate_runtime_system as system;
+
+#[cfg(feature = "std")] extern crate serde;
+
+//extern crate safe_mix;
+
+use rstd::prelude::*;
+use rstd::marker::PhantomData;
+use primitives::{Zero, One, Headery, Blocky, Checkable, Applyable, CheckEqual, Hashing, Executable};
+use codec::Slicable;
+//use runtime_support::{Hashable, StorageValue, StorageMap};
+
+//use primitives::{AuthorityId, Hash, BlockNumber, Header, Log};
+//use primitives::block::{generic, Number as BlockNumber, Header, Log};
+
+//use safe_mix::TripletMix;
+
+pub struct Executive<
+	Unchecked,
+	Checked,
+	System,
+	Block,
+	Finalisation,
+>(PhantomData<(Unchecked, Checked, System, Block, Finalisation)>);
+
+impl<
+	Unchecked: Checkable<
```

Comment:
this would go more nicely in a where clause

Reply:
perhaps

Reply:
ok so i tried that because, ya know, i want to be an obedient little rustacean. here's what there was before:

```
impl<
	Unchecked: Checkable<
		CheckedType = Checked
	> + PartialEq + Eq + Clone + Slicable,
	Checked: Applyable<
		IndexType = System::Index,
		AccountIdType = System::AccountId
	>,
	System: system::Trait,
	Block: Blocky<
		Extrinsic = Unchecked,
		Header = System::Header
	>,
	Finalisation: Executable,
> Executive<Unchecked, Checked, System, Block, Finalisation> {
```

and here's the after:

impl<
	Unchecked,
	Checked,
	System,
	Block,
	Finalisation,
> Executive<Unchecked, Checked, System, Block, Finalisation> where
	Unchecked: Checkable<
		CheckedType = Checked
	> + PartialEq + Eq + Clone + Slicable,
	Checked: Applyable<
		IndexType = System::Index,
		AccountIdType = System::AccountId
	>,
	System: system::Trait,
	Block: Blocky<
		Extrinsic = Unchecked,
		Header = System::Header
	>,
	Finalisation: Executable,
{
```

here's the statistics of that change:

- 6 extra lines of code;
- 1 extra key word used (`where`)
- 5 extra (entirely redundant) syntactic dependencies

the syntactic dependencies are generally bad for maintainability. it would be extremely difficult to make an argument for there being any additional readability.